pymc_experimental/statespace/filters/distributions.py

```diff
@@ -18,6 +22,64 @@
 )
 
 
+def make_signature(sequence_names):
+    states = "s"
+    obs = "p"
+    exog = "r"
+    time = "t"
+    state_and_obs = "n"
+
+    matrix_to_shape = {
+        "x0": (states,),
+        "P0": (states, states),
+        "c": (states,),
+        "d": (obs,),
+        "T": (states, states),
+        "Z": (obs, states),
+        "R": (states, exog),
+        "H": (obs, obs),
+        "Q": (exog, exog),
+    }
+
+    for matrix in sequence_names:
+        base_shape = matrix_to_shape[matrix]
+        matrix_to_shape[matrix] = (time,) + base_shape
+
+    signature = ",".join(["(" + ",".join(shapes) + ")" for shapes in matrix_to_shape.values()])
+
+    return f"{signature},[rng]->[rng],({time},{state_and_obs})"
+
+
+class MvNormalSVDRV(MvNormalRV):
+    name = "multivariate_normal"
+    signature = "(n),(n,n)->(n)"
+    dtype = "floatX"
+    _print_name = ("MultivariateNormal", "\\operatorname{MultivariateNormal}")
+
+
+class MvNormalSVD(MvNormal):
+    """Dummy distribution intended to be rewritten into a JAX multivariate_normal with method="svd".
+
+    A JAX MvNormal robust to low-rank covariance matrices
+    """
+
+    rv_op = MvNormalSVDRV()
+
+
+@jax_sample_fn.register(MvNormalSVDRV)
+def jax_sample_fn_mvnormal_svd(op, node):
+    def sample_fn(rng, size, dtype, *parameters):
+        rng_key = rng["jax_state"]
+        rng_key, sampling_key = jax.random.split(rng_key, 2)
+        sample = jax.random.multivariate_normal(
+            sampling_key, *parameters, shape=size, dtype=dtype, method="svd"
```

Comment:
Interesting.

More importantly, aren't you making JAX non-optional with the imports?

Reply:
Yes, good catch. I'll import inside the rewrite.

Reply:
I think importing the dispatch already forces jax to be importable. Usually I have had to have a try except import of jax

Reply:
Oh at the top of the file you mean?

Reply:
let me know if I did it right

Reply:
If you can import the package or the statespace submodule without jax istalled you did it right